source/ceylon/time/timezone/parser/iana/parseMonth.ceylon

```diff
@@ -0,0 +1,11 @@
+import ceylon.time.base {
+    months,
+    Month
+}
+
+shared Month parseMonth(String month) {
+    "Invalid Month for parse in timeZone"
+    assert(exists currentMonth = months.all.find((Month elem) 
+        => elem.string.lowercased.startsWith(month.trimmed.lowercased)));
```

Comment:
As a side note though, we might need to reconsider this implementation from the performance perspective in the future.